--- src/VitaTouch.cpp
+++ src/VitaTouch.cpp
@@ -1,6 +1,6 @@
 #include "VitaTouch.hpp"
 #include <stdlib.h>
-
+#include <psp2/kernel/processmgr.h>

 VitaTouch::VitaTouch(){
	emptyClickPoint.x = emptyClickPoint.y = 0;
@@ -19,6 +19,10 @@
	for(port = 0; port < SCE_TOUCH_PORT_MAX_NUM; port++){
		sceTouchPeek(port, &touch[port], 1);
	}
+	if (touch_old[SCE_TOUCH_PORT_FRONT].reportNum == 0 && touch[SCE_TOUCH_PORT_FRONT].reportNum >= 1) {
+		touchDownTime = sceKernelGetProcessTimeWide();
+	}
+
	if (touch_old[SCE_TOUCH_PORT_FRONT].reportNum >= 1 && touch[SCE_TOUCH_PORT_FRONT].reportNum == 0){
		if(!scrolling){
			lastClickPoint.x = touch_old[SCE_TOUCH_PORT_FRONT].report[0].x / 2;  // 1920 touch => 960 display
@@ -28,6 +32,7 @@
			lastTouchPoint = emptyClickPoint;
			scrolling = false;
			clicking = true;
+			touchDuration = sceKernelGetProcessTimeWide() - touchDownTime;
		}else{
			lastClickPoint = emptyClickPoint;
			lastTouchPoint = emptyClickPoint;
